bus/bus.go

```diff
@@ -1810,64 +1810,54 @@ func (b *bus) metricsHandlerPUT(jc jape.Context) {
 
 func (b *bus) metricsHandlerGET(jc jape.Context) {
 	key := jc.PathParam("key")
+
+	// parse mandatory query parameters
 	var err error
+	var start time.Time
+	var n uint64
+	var interval time.Duration
+	if jc.DecodeForm("start", (*api.TimeRFC3339)(&start)) != nil {
+		return
+	} else if jc.DecodeForm("n", &n) != nil {
+		return
+	} else if jc.DecodeForm("interval", (*api.DurationMS)(&interval)) != nil {
+		return
+	}
+
+	// parse optional query parameters
 	switch key {
 	case api.MetricContract:
 		var metrics []api.ContractMetric
 		var opts api.ContractMetricsQueryOpts
-		if jc.DecodeForm("after", (*api.TimeRFC3339)(&opts.After)) != nil {
-			return
-		} else if jc.DecodeForm("before", (*api.TimeRFC3339)(&opts.Before)) != nil {
-			return
-		} else if jc.DecodeForm("fcid", &opts.FCID) != nil {
+		if jc.DecodeForm("fcid", &opts.FCID) != nil {
```

Comment:
I think we would usually go for `contractID` when it is user facing right?
Also `host` should be `hostKey` I think